Disable some hosting tests failing on CI

radical opened the pull request:
Issues: https://github.com/dotnet/aspire/issues/4650 and https://github.com/dotnet/aspire/issues/4651

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4653)

radical committed:
Disable some hosting tests failing on CI

eerhardt commented on tests/Aspire.Hosting.Tests/DistributedApplicationTests.cs:
```diff
@@ -171,6 +171,7 @@ public Task AfterEndpointsAllocatedAsync(DistributedApplicationModel appModel, C
     }
 
     [Fact]
+    [ActiveIssue("https://github.com/dotnet/aspire/issues/4651", typeof(PlatformDetection), nameof(PlatformDetection.IsRunningOnCI))]
```
Why isn't this

```suggestion
    [ActiveIssue("https://github.com/dotnet/aspire/issues/4651")]
```

?

If a test is flaky, I don't want it to fail on my machine locally only to realize it might be broken and not due to my changes.

radical replied:
This was working fine earlier on local runs, and CI. But started to fail around the time we enabled docker based tests to run on the build machine. But I think this affected by other tests that run before this.

Instead of disabling this one, I will revert all the tests in `tests/Aspire.Hosting.Tests/DistributedApplicationTests.cs` which had `[LocalOnly*]` back to being local-only. This is what @mitchdenny suggested in https://github.com/dotnet/aspire/pull/4142#issuecomment-2188780945 too.

radical replied:
I retained the attributed as-is on this test because it was also marked `LocalOnly*` earlier.

radical replied:
@eerhardt I haven't heard from anyone about this test failing locally. But I can disable it for local too, if that's preferable.

radical replied:
Merging this to get a green CI. But I can make this change in a follow up PR if you want.

eerhardt replied:
> I haven't heard from anyone about this test failing locally.

My guess is that it isn't getting run as often locally as it is in CI.

My thinking is that CI machines are just normal computers. If we have races / flaky failures in our tests on CI, chances are we do locally too.

radical committed:
Aspire.Hosting.Testsing.DistributedApplicationTests - revert tests to run local only

Some tests were enabled to run on CI in
fb97edac64b8541684e4268a787892e1deca20a7 but they might be responsible
for some intermittent failures. This commit disables them till the issue
can be fixed.
